### DISCUSSION TITLE
Required kernel patches for maximum throughput

### DISCUSSION THREAD

**Discussion by Freshdouble** [svpcom/wfb-ng#86] [Questions]
I just wanted to ask what kernel patches and what kernel config options are necessary for maximum data throughput with bidirectional communication.

The only patches in the patch directory for bidirectional data flow are patches for the ralink chipset that only affect the maximum output power of those sticks as far as i understand.

I want to use the newest Raspberry Pi OS and i got a working video but only with 500k Bitrate. All other bitrates above this result in a unstable Video and i think this is an issue with the throupghput of the wifibroadcast link.

**Comment by svpcom** [comment]
For modern kernels (4.x and later) you may only need patch to disable crda (to output max power). What cards (what chipset) do you use? First try to insert them to linux desktop/laptop. Raspberry pi has known issues with usb power. Also add `net.core.bpf_jit_enable = 1` to /etc/sysctl.conf
